src/inverse-kinematics/CMakeLists.txt

```diff
@@ -37,6 +37,7 @@ target_include_directories(${libraryname} SYSTEM PRIVATE ${EIGEN3_INCLUDE_DIR})
 target_link_libraries(${libraryname} idyntree-core idyntree-high-level ${IPOPT_LIBRARIES})
 
 target_compile_definitions(${libraryname} PRIVATE ${IPOPT_DEFINITIONS})
+target_compile_definitions(${libraryname} PRIVATE IDYNTREE_USES_IPOPT)
```

Comment:
As you can see from the GitHub Actions failures, you need to also modify the CMakeLists.txt of this library to link IPOPT, pass the `IDYNTREE_USES_IPOPT` definition, add the private sources that depend on IPOPT only if `IDYNTREE_USES_IPOPT ` is set to FALSE, similarly to what is done for assimp in the solid-shapes lib, see https://github.com/robotology/idyntree/blob/v1.0.1/src/solid-shapes/CMakeLists.txt . Note that as we require CMake 3.5 now, if necessary you can also use `target_sources` command, see https://github.com/robotology/idyntree/blob/v1.0.1/src/solid-shapes/CMakeLists.txt .